plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -1137,4 +1245,381 @@ function OPDSBrowser:downloadDownloadList()
+                                entry.file = copy_download_path
+                                table.insert(dl_list, entry)
+                            end
+                            Trapper:wrap(function()
+                                dismissable_download()
+                            end)
+                        end
+                    },
+                },
+            },
+        }
+        UIManager:show(textviewer)
+    end
+end
+
+-- Set string for desired filetypes
+function OPDSBrowser:setSyncFiletypes(filetype_list)
+    local input = self.settings.filetypes
+    local dialog
+    dialog = InputDialog:new{
+        title = _("File types to sync"),
+        description = _("A comma separated list of desired filetypes"),
+        input_hint = _("epub, mobi"),
+        input = input,
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    id = "close",
+                    callback = function()
+                        UIManager:close(dialog)
+                    end,
+                },
+                {
+                    text = _("Save"),
+                    is_enter_default = true,
+                    callback = function()
+                        local str = dialog:getInputText()
+                        self.settings.filetypes = str ~= "" and str or nil
+                        self._manager.updated = true
+                        UIManager:close(dialog)
+                    end,
+                },
+            },
+        },
+    }
+    UIManager:show(dialog)
+    dialog:onShowKeyboard()
+end
+
+function OPDSBrowser:checkSyncDownload(idx)
+    if self.settings.opds_sync_dir then
+        self.sync = true
+        if idx then
+            self:fillPendingSyncs(self.servers[idx-1]) -- First item is "Downloads"
+        else
+            for _, item in ipairs(self.servers) do
+                if item.sync then
+                    self:fillPendingSyncs(item)
+                end
+            end
+        end
+        if #self.pending_syncs > 0 then
+            Trapper:wrap(function()
+                self:downloadPendingSyncs()
+            end)
+        else
+            UIManager:show(InfoMessage:new{
+                text = _("Up to date!"),
+            })
+        end
+        self.sync = false
+    else
+        UIManager:show(InfoMessage:new{
+            text = _("Please select a folder for sync downloads first"),
+        })
+    end
+end
+
+function OPDSBrowser:updateFieldInCatalog(item, name, value)
+    item[name] = value
+    self._manager.updated = true
+end
+
+function OPDSBrowser:setMaxSyncDownload()
+    local current_max_dl = self.settings.opds_sync_max_dl or 50
+    local spin = SpinWidget:new{
+        title_text = "Set maximum sync size",
+        info_text = "Set the max number of books to download at a time",
+        value = current_max_dl,
+        value_min = 10,
```

Comment:
Let if be 1, if a user wants to just test.
Also you can add `default_value = 50`.